src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -1180,7 +1178,9 @@ private void ImportBox(int token)
                     }
                 }
 
-                ThrowHelper.ThrowInvalidProgramException();
+                // If this is a byref-like type, only the above operations are permitted.
```

Comment:
This works only if we recognize all patterns that are recognized by the JIT. I see RyuJIT recognizing more patterns than what we do here - e.g. something with LDNULL: https://github.com/dotnet/runtime/blob/f23d779533aefb4537ffa4632116f277b14da179/src/coreclr/jit/importer.cpp#L3216